### DISCUSSION TITLE
TetrahedronElement.GetLocalInternalStress() + Robot Structural Analysis Comparison

### DISCUSSION THREAD

**Discussion by Murtasgh** [BriefFiniteElementNet/BriefFiniteElement.Net#152] [Q&A]
Hi,

I am aware that the TetrahedronElement is not fully implemented and may contain some bugs. For this reason I was comparing the output from BFE.NET with Robot Structural Analysis and the deflections are pretty much the same.

However, I'm having some difficulties extracting the stresses on TetrahedronElement by using GetLocalInternalStress(). I keep getting an IndexOutOfRangeException which seems to be because the res[,] variable is always [6x1] when I guess it should be at least [3x3] (see screenshot attached).

Are there any example on how GetLocalInternalStress() should be used to avoid this?
  
![image](https://user-images.githubusercontent.com/40386478/225401339-e43d9536-d2dd-49eb-beab-d6ae532c4c5b.png)

Many thanks for all the help in advance.

**Comment by epsi1on** [comment]
Hi there,
I need to check, seems the matrix dimensions are mismatch.
Formula for stress tensor would be D*B*U, but in this case...
Will update soon...
Also it's cool to hear the displacements are similar with robot structural, it could be a success validation of the library...
edit: for my own reference this is a formula for internal force of tetrahedron element: [link](http://what-when-how.com/the-finite-element-method/fem-for-3d-solids-finite-element-method-part-1/)

**Comment by epsi1on** [comment]
fixed in 89e312488e8e0a8e533bda84bce844cc90bf04ec,
can you please do a check?

**Reply by Murtasgh** [reply]
That's perfect, I'll do a check as soon as possible. 

In a similar note, I was validating the TriangularShell elements but I'm having some difficulties with the LocalX and LocalY coordinates of the GetInternalForce method. Would you be able to confirm what coordinates would correspond to points 1, 2 and 3 of each triangle?

Thanks!

**Reply by epsi1on** [reply]
GetInternalForce usualy uses isoparametric coordinates. it is not a fixed point. do you mean integration points?

**Comment by epsi1on** [comment]
inactivity
